src/main/java/org/jitsi/videobridge/octo/OctoPacket.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright @ 2015-2017 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.videobridge.octo;
+
+import org.jitsi.service.neomedia.*;
+import org.jitsi.util.*;
+
+/**
+ * A utility class which handles the on-the-wire Octo format. Octo encapsulates
+ * its payload (RTP, RTCP, or anything else) in an 8-byte header:
+ * <pre>{@code
+ *  0                   1                   2                   3
+ *  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |R| M | S | res |              Conference ID                    |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |                         Endpoint ID                           |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * }</pre>
+ * R: Source-is-a-relay flag. 1 if the source of the packet is a relay, and
+ * 0 if it is an endpoint (bridge).
+ * <p/>
+ * M: media type (audio, video, or data).
+ * <p/>
+ * S: Simulcast layer ID.
+ *
+ * @author Boris Grozev
+ */
+public class OctoPacket
```

Comment:
i'm wondering why go with the static reader methods as opposed to `OctoPacket` deriving from `DatagramPacket` (or one of the other packet types) and having the members parse the internal buffer for the values?

not clear to me that it would necessarily better, but was wondering if you considered it.

Reply:
I haven't considered it before. We operate on the DatagramPacket instances coming from libjitsi, so we would have to create a new wrapping instance if we wanted to do this (or am I missing your point?).

Reply:
ok, my comment was based on the assumption that, wherever we created the `DatagramPacket` (or where we would normally create a RawPacket or whatever) we would instead just create an `OctoPacket`, but are you saying that there is no place where that would happen?

Reply:
Correct. The DatagramPacket instance comes from libjitsi, which doesn't know about Octo. We intercept the packet and add/remove Octo headers in jitsi-videobridge